hatch/api_client.py

```diff
@@ -0,0 +1,53 @@
+import httpx
+from fastapi import HTTPException, Response
+
+from hatch import config
+
+
+# using a module level client should keep the connection open to job-server
+client = httpx.Client(
+    base_url=config.API_SERVER, headers={"Authorization": config.BACKEND_TOKEN}
+)
+
+
+def create_release(workspace, release, user):
+    """API call to job server to create a release."""
+    response = client.post(
+        url=f"/api/v2/releases/workspace/{workspace}",
+        content=release.json(),
+        headers={"OS-User": user},
+    )
+    if response.status_code != 201:
+        raise proxy_httpx_error(response)
+
+    return proxy_httpx_response(response)
+
+
+def _proxy_headers(orig_headers):
+    headers = orig_headers.copy()
+    for header in ["Connection", "Server"]:
+        headers.pop(header, None)
+    headers["Via"] = config.SERVER_HOST
+    return headers
+
+
+def proxy_httpx_response(response):
+    """Take an upstream httpx response and convert to a proxied FastAPI Response."""
```

Comment:
I'm not sure I understand what the proxied part of this is doing/for, could you explain it further for me?

Reply:
So we have a `httpx.Response` object, and we want to send that to the client. So we copy across status/body/headers to a `fastapi.Response` so that we can send it.

However, when proxying http responses, there are certain headers that should not be copied from the upstream, particularly `Connection`, as that is hop-specific. fastapi will add it's own `Server` and `Connection` headers as part of it's normal set up.